wls-gui-wahllokalsystem/src/components/common/inputs/BaseDateInput.vue

```diff
@@ -0,0 +1,39 @@
+<template>
+  <v-text-field
+    :model-value="formatedDate"
+    type="date"
+    :rules="[REQUIRED]"
+    label="Datum"
+    clearable
+    @update:model-value="onModelValueChanged"
+    @clear="onClearClicked"
+  />
+</template>
+
+<script setup lang="ts">
+import { computed } from "vue";
+import { VTextField } from "vuetify/components";
+
+import { useDateTimeFormatter } from "@/composables/common/dateTimeFormatter.ts";
+import { REQUIRED } from "@/util/rules.ts";
+
+const { toIsoDate } = useDateTimeFormatter();
+
+const model = defineModel<Date>();
+
+const formatedDate = computed(() =>
+  model.value ? toIsoDate(model.value) : null
+);
+
+function onClearClicked() {
+  model.value = undefined;
+}
+
+function onModelValueChanged(newValue?: string) {
```

Comment:
warum optional wenn du danach nochmal prüfst ob der parameter da ist?
```suggestion
function onModelValueChanged(newValue: string) {
```

Reply:
Streng genommen müsste es auch `null` umfassen. `null` kommt wenn ich das Feld cleare.

Entsprechend muss der Parameter als optional definiert werden. Für die weitere Implementierung ist es auch relevant. Nur wenn es wirklich ein String ist kann ich daraus ein Date erzeugen. Ohne die deklaration als optionaler Parameter würde der Check keinen Sinn ergeben.

Reply:
bei clear sollte die funktion doch garnicht aufgerufen werden. wenn nur null kommen kann sollte es `onModelValueChanged(newValue: string | null)` heißen

Reply:
Funktion wurde angepasst: c9986c4e283d5bac358c77be6dee2a0c02ef0349